src/robotFactory.js

```diff
@@ -1,10 +1,87 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    if (Object.keys(coords).length === 0) {
+      this.coords = {
+        x: coords.x || 0,
+        y: coords.y || 0,
+      };
+    };
```

Comment:
Additional check is redundant.
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
```